test/caf_co_max_test.f90

```diff
@@ -105,12 +105,11 @@ function max_elements_in_3D_string_arrays() result(result_)
       type(result_t) result_
       character(len=*), parameter :: script(*) = ["To be ","or not","to    ","be.   "]
       character(len=len(script)), dimension(2,1,2) :: scramlet, co_max_scramlet
-      integer i, cyclic_permutation(size(script))
+      integer i, cyclic_permutation(size(script)), me
 
-      associate(me => this_image())
-        associate(cyclic_permutation => [(1 + mod(i-1,size(script)), i=me, me+size(script) )])
-          scramlet = reshape(script(cyclic_permutation), shape(scramlet))
-        end associate
+      call prif_this_image_no_coarray(this_image=me)
+      associate(cyclic_permutation => [(1 + mod(i-1,size(script)), i=me, me+size(script) )])
+        scramlet = reshape(script(cyclic_permutation), shape(scramlet))
```

Comment:
@rouson CI shows the scramlet test is now failing.

I suspect the problem is the previous incorrect use of `this_image()` was setting `me==1` for all images, but now we actually set it to the image index, and the code fails

Reply:
One (probably unrelated) bug that jumps out at me here is the code below that does:
```fortran
call prif_co_max(co_max_scramlet, result_image=1)
```
Which leaves `co_max_scramlet` undefined on images other than 1, but the code later inspects the value of `co_max_scramlet` on all images, leading to undefined behavior.

We probably need to drop the `result_image` argument to provide the result to all images.

Reply:
@ktras the comment above by @bonachea likely points us to the problem that the cause of the `prif_co_max` unit test failure.